autode/bond_rearrangement.py

```diff
@@ -10,6 +13,211 @@
+                        list(known_bbond_types.items()),
+                        list(known_fbond_types.items()),
+                    )
+                )
+            return None
+
+        type_combs = list(
+            combinations_with_replacement(common_types, self._n_extra_pair)
+        )
+
+        for comb in type_combs:
+            this_bbond_types = known_bbond_types.copy()
+            this_fbond_types = known_fbond_types.copy()
+            for key in comb:
+                this_bbond_types[key] = this_bbond_types.get(key, 0) + 1
+                this_fbond_types[key] = this_fbond_types.get(key, 0) + 1
+            self._movesets.append(
+                (
+                    list(this_bbond_types.items()),
+                    list(this_fbond_types.items()),
+                )
+            )
+
+        self._sanitise_movesets()
+        return None
+
+    def _sanitise_movesets(self):
+        """
+        Check that all the movesets are actually appropriate and
+        are not forming/breaking more bonds than is actually possible
+        """
+        new_movesets = []
+        for moveset in self._movesets:
+            is_valid = True
+            for bbond_type, num in moveset[0]:
+                if len(self._rct_bond_dict[bbond_type]) < num:
+                    is_valid = False
+                    break
+            for fbond_type, num in moveset[1]:
+                if len(get_fbonds(self._reactant.graph, fbond_type)) < num:
+                    is_valid = False
+                    break
+            if is_valid:
+                new_movesets.append(moveset)
+
+        self._movesets = new_movesets
+        return None
+
+    def _graph_edits(
+        self,
+        moveset: tuple[list[tuple], list[tuple]],
+        bbonds: Optional[tuple] = None,
+        fbonds: Optional[tuple] = None,
+        counter: int = 0,
+    ) -> Iterator[list[tuple]]:
+        """
+        Generator that yields all possible forming and breaking bonds
+        based on the allowed graph moves defined in moveset. Only the
+        moveset should be supplied, other arguments are used to keep
+        track of the internal state of the generator.
+
+        Args:
+            moveset: A tuple of two lists, each of the type [("CH", 2), ...],
+                     first being the types and numbers of bonds to be broken
+                     and the second being those to be formed
+            bbonds: (State variable) tuple of bonds broken so far
+            fbonds: (State variable) tuple of bonds formed so far
+            counter: (State variable) number of bond types operated
+
+        Yields:
+            (list[tuple]): A list with two items, the first being
+                        the bonds being broken, and the second being
+                        the bonds being formed
+        """
+        assert self._reactant.graph is not None
+        assert self._rct_bond_dict is not None
+        assert len(moveset[0]) + len(moveset[1]) != 0
```

Comment:
what happens if you call this from the top level with a `A -> A` identity reaction?

Reply:
I mean like `C[C+]([H])[H]` -> `C[C+]([H])[H]`. basically I'm wondering how this gets hit